apps/frontend/src/pages/Homepage.tsx

```diff
@@ -1,5 +1,246 @@
+import {
+  Grid,
+  Box,
+  Typography,
+  Pagination,
+  FormControl,
+  Select,
+  MenuItem,
+  SelectChangeEvent,
+  FormHelperText,
+} from "@mui/material";
+import { useStyles } from "../styles/pageStyles";
+import SearchBar from "../components/searchBar.tsx";
+import ListingCard from "../components/listingCard.tsx";
+import { useState, useEffect } from "react";
+import * as React from "react";
+
+interface ListingObject {
+  listingID: string;
+  sellerID: string;
+  sellerName: string;
+  title: string;
+  description: string;
+  price: number;
+  dateCreated: string;
+  imageUrl: string;
+}
+
 const Homepage = () => {
-  return <h1>Homepage</h1>;
+  const classes = useStyles();
+
+  // Listings per page
+  const listingsPerPage = 6;
+
+  const [listingObjects, setListingObjects] = useState<ListingObject[]>([]);
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [searchPerformed, setSearchPerformed] = useState(false);
+  const [sortBy, setSortBy] = React.useState<string>(""); // Add this state
+
+  const handleSortBy = (event: SelectChangeEvent<string>) => {
+    setSortBy(event.target.value as string);
+  };
+
+  // Calculate the total number of pages
+  const totalPages = Math.ceil(listingObjects.length / listingsPerPage);
+
+  // Get the listings for the current page
+  const displayedListings = listingObjects.slice(
+    (currentPage - 1) * listingsPerPage,
+    currentPage * listingsPerPage,
+  );
+
+  const handlePageChange = (
+    _event: React.ChangeEvent<unknown>,
+    page: number,
+  ) => {
+    setCurrentPage(page);
+  };
+
+  const handleSearch = () => {
+    //FETCH API DATA??
+    // UPDATE LISTING OBJECTS
+    setSearchPerformed(true);
+  };
+
+  const fetchListings = async () => {
+    setListingObjects([
```

Comment:
this mock data object should be defined in a separate file

Reply:
moved the mocked data to a different file